### ISSUE TITLE
CodeCoverage.xml fails to include all files from -CodeCoverageFolder 

### ISSUE THREAD

**Issue by PwshAddict** [pester/AzureDevOpsExtension#38]
### Where are you running it?
*You can run the extensions in the cloud or on-premise*
- Azure DevOps Service (VSTS)

### Version of Extension/Task
10.1.17

### Expected behaviour and actual behaviour
I expected all .ps1 files to be used for the codecoverage.xml file but only the ones in the ./tests folder show results.

### Steps to reproduce the problem
Set `-CodeCoverageFolder .` or to the base folder
├───.vscode
├───docs
│   └───images
├───ModuleFolder
│   ├───private
│   └───public
├───TestResults
└───Tests

**azure-pipelines.yml** file:
```
 - task: Pester@10
      displayName: Pester Test Runner
      inputs:
        TestFolder: './tests'
        resultsFile: '$(Common.TestResultsDirectory)/Test-Pester.XML'
        CodeCoverageOutputFile: '$(Common.TestResultsDirectory)/CC-Pester.XML'
        usePSCore: ${{ parameters.pscore }}
        CodeCoverageFolder: '$(Build.Repository.LocalPath)/'
        PesterVersion: OtherVersion
        preferredPesterVersion: 5.0.2
        additionalModulePath: '$(System.DefaultWorkingDirectory)/**/*.psm1'
```
### Findings during testing

**$pesterConfig** is correct and **$pesterconfig.CodeCoverage.Path** returns all the files.
```
$PesterConfig['CodeCoverage'] = $CodeCoverage

if (-not([String]::IsNullOrWhiteSpace($ScriptBlock))) {
    $ScriptBlockObject = [ScriptBlock]::Create($ScriptBlock)

    $ScriptBlockObject.Invoke()
}

$result = Invoke-Pester -Configuration  ([PesterConfiguration]$PesterConfig)
```
However, when the configuration object is created, **([PesterConfiguration]$PesterConfig).CodeCoverage.Path.Value** is an empty array.

**Comment by ChrisLGardner** [comment]
Thanks, I know why this is happening (can't remember if there's a github issue on the main repo for it) and know how I'll fix it here. I'll add it to my list to look at as soon as I get some time.

**Comment by PwshAddict** [comment]
I was able to resolve it by creating the [pesterconfiguration] object first and then update the settings along the way.  It's on my fork at the moment.  If this how you were thinking, I can create a PR for it.

**Comment by ChrisLGardner** [comment]
Yeah, that's basically the issue. It doesn't cast the nested hashtables to the right thing for some reason. If you've got a PR then that'd be welcome.

**Comment by PwshAddict** [comment]
PR sent and added you as a collaborator on my repo.  Didn't see a way to make you an owner.

**Comment by ChrisLGardner** [comment]
Fixed in #39. Thanks for the help there.
